Medium level solution

Muhab2001 opened the pull request:
my solution for the medium-level challenge using Dart. 

sorry if there was a way to submit a `.dart` file instead of this mess

Muhab2001 committed:
Medium level solution

my solution for the medium level challenge using dart

Muhab2001 committed:
fixing extension

mohalobaidi commented:
I'll check your submission tomorrow ❤️

mohalobaidi commented:
Sorry for the late response, there is a logical error in intializeBoundary method

Looking forward for your edited version

Muhab2001 committed:
edited solution

Sorry for the very late submission. I thought someone already solved it and didn't bother to send my corrected code
